### PR TITLE
[SQL] Many fixes and improvements overall

### PR DESCRIPTION
**Pull request by azizk**
Hey!

My main goal was to fix the way the word-boundary escape `\b` is used throughout the syntax file. I did much more besides that.
So I wanted, for example, `1IN` to be two tokens, and I assumed that every engine recognizes them as such, but I just tested with PSQL and MySQL and it looks like it's 2 tokens for PSQL and 1 for MySQL.

```sql
-- This is valid in PSQL and returns `true`. MySQL thinks it's an undefined function called "1IN".
SELECT 1IN(1,2,3);
```

I haven't looked at what the standard says yet. Regardless, I think my changes are pretty useful as they are and the thing above can be reverted by adding a `\b` back to the `match: \d+` context. My last commit is solely for formatting. Hope you like it! :+1:

### CODE REVIEW

**Comment by deathaxe** [comment]
Many thanks to your contribution. I wonder whether we can incooperate any valuable changes into #3046 as it already comes a long with a lot of improvements especially with regards to syntaxes for different SQL dialects based on one "root-syntax".

**Comment by azizk** [comment]
Oh dear, I should have done a search before doing this. :smile:
I'll check it out later...

**Comment by jrappen** [comment]
Yes, as @deathaxe said, please take a look at @keith-hall 's #3046. As you can see, his branch for that PR is <https://github.com/forkeith/Packages/tree/sql> which you can use for a diff and/or possible PRs to target and be included in #3046.

**Comment by azizk** [comment]
Thanks @jrappen! Yeah I checked it out and it's looking great. There are some things I'd do differently (the leading `\b`) and I could also contribute some more stuff, which isn't included in this PR (mostly PostgreSQL). Hey @keith-hall, if you're interested I could contribute and send a PR to your branch. :+1:

Nevertheless, do you all think it's still worthwhile to merge this in? I don't mind if it's overwritten later by Keith's work, but until then these changes may be useful and deployed to users earlier? :smiley:

File: SQL/SQL.sublime-syntax [hunk 1]
```diff
@@ -8,10 +8,7 @@ file_extensions:
   - ddl
   - dml
 
-first_line_match: |-
-  (?xi:
-    ^ \s* -- .*? -\*- .*? \bsql\b .*? -\*-  # editorconfig
-  )
+first_line_match: (?xi) ^ \s* -- .*? -\*- .*? \bsql\b .*? -\*-  # editorconfig
```
Comment (keith-hall) [inline SQL/SQL.sublime-syntax:11]: arguably, keeping this on multiple lines is better for consistency with other syntaxes https://github.com/sublimehq/Packages/commit/3629e8bce1d25fbbc66528ed212f94597cbb8800

**Comment by keith-hall** [comment]
a PR to my branch improving PostgreSQL would be very welcome :)
I'm not sure I agree with the blanket removal of leading `\b` though. It would make sense to do this for the PSQL variant, sure, but it'd be nice to keep it for the others, so even though merging this PR early could be beneficial for PSQL users, it may be a regression for all others.

**Comment by azizk** [comment]
@keith-hall Alright, gonna send some commits then. :smiley:
So regarding the leading word anchor `\b`, afaics it's not necessary. It's only needed if you don't consume all the word characters (the way it is now), but if you do then using a leading `\b` everywhere is pretty redundant. Or maybe I'm overlooking something? And in case you want to recognise something like `1IN` as a single token, like I described in my PR comment, then adding a *trailing* `\b` to the number regexp is all there is to make it work.

File: SQL/SQL.sublime-syntax [hunk 2]
```diff
@@ -9,9 +9,8 @@ file_extensions:
   - dml
 
 first_line_match: |-
-  (?xi:
-    ^ \s* -- .*? -\*- .*? \bsql\b .*? -\*-  # editorconfig
-  )
+  (?xi)
```
Comment (jrappen) [inline SQL/SQL.sublime-syntax:12]: please revert this change

Comment (azizk) [reply SQL/SQL.sublime-syntax:12]: Ok. Thought it's consistent with the rest of the formatting I did...

Comment (jrappen) [reply SQL/SQL.sublime-syntax:12]: You should never use this style for variables, as it'll have unintended consequences when using them.

Comment (azizk) [reply SQL/SQL.sublime-syntax:12]: You're right, for variables it's probably not the best. But `first_line_match` isn't really a variable, right?
Now that you're saying that, I'll have to check whether this is an issue in ElixirSyntax. Maybe I was lucky and didn't run into issues with `(?x)` in variables...

Comment (jrappen) [reply SQL/SQL.sublime-syntax:12]: Sure, then let me rephrase this: Explicit is better than implicit.

**Comment by jrappen** [comment]
What's the current status with this PR and the request to add your changes to Keith's PR?

So far you have only renamed some stuff on Keith's branch and not added any meaningful changes from this PR to his.

**Comment by azizk** [comment]
@jrappen From my end the PR here can go in. Keith is concerned about the leading `\b` however. I asked about it in my previous comment...
I know I only contributed minor things on his branch. I haven't committed my other progress yet. Just had other priorities in the meantime.

**Comment by azizk** [comment]
I just rebased the branch. If the PR is still useful please merge, otherwise let's close this. I'm lacking the time and resources to continue with this...

**Comment by azizk** [comment]
I'm not interested in this anymore. If you still want the changes, feel free to merge or cherry-pick them. :+1:
